flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorTest.java

```diff
@@ -2896,7 +2896,8 @@ private TaskExecutorTestingContext createTaskExecutorTestingContext(
                                 .setTaskStateManager(stateStoresManager)
                                 .setTaskChangelogStoragesManager(changelogStoragesManager)
                                 .build(),
-                        HEARTBEAT_SERVICES,
+                        // prevent heartbeat timeouts from failing the tests focused on other aspects
+                        new HeartbeatServices(60_000L, 60_000L),
```

Comment:
Would it make sense to disable heartbeats entirely instead of relying on higher timeouts?

Reply:
IIUC, this is currently not possible, and adding this logic seemed too much for this case. WDYT?

Reply:
AFAIK it's not; it should be straightforward to add it.

Since we already have the `NoOpHeartbeatManager` implementation, I think the only thing missing is extracting `HeartbeatServices` into an interface (two methods) and introducing the `NoOpHeartbeatServices` implementation (both methods would simply return `NoOpHeartbeatManager`).

Having this in place would be helpful for other tests as well.